client/src/App.js

```diff
@@ -67,6 +65,28 @@ const StyledLink = styled(Link)`
 
 class App extends React.Component<Props, State> {
 
+  state = {
+    filters: {
+      name: null,
+      cuisine_id: null,
+      rating: null,
+      max_delivery_time_minutes: null
+    }
+  }
+
+  constructor(props){
+    super(props)
+    this.searchByName = this.query.bind(this, 'name')
```

Comment:
Not too fond of this style, but its quite clever!
could be cause im just not used to this.

Reply:
Yee, it's always a dilemma. It is a good was to compose functions though.